monitoring/uss_qualifier/scd/executor/runner.py

```diff
@@ -0,0 +1,105 @@
+
+    def run_automated_test(self, dry=False):
+        """Runs the automated test. If dry is set to True, no request is submitted and results of requests are set to DryRun.
+           This allows to introspect a test plan without dependencies."""
+
+        for step in self.automated_test.steps:
+            print('[SCD]   Running step {}'.format(step.name))
+            self.execute_step(step, dry=dry)
+
+    def teardown(self, dry=False):
+        """Delete resources created by this test runner."""
+        print("[SCD]   Teardown {}".format(self.automated_test.name))
+        for role, target in self.targets.items():
+            target.delete_all_flights(dry=dry)
+
+    def execute_step(self, step: TestStep, dry=False):
+        target = self.get_target(step)
+        if target is None:
+            # TODO implement reporting
+            self.print_targets_state()
+            raise RuntimeError("[SCD] Error: Unable to identify the target managing flight {}".format(
+                step.inject_flight.name if 'inject_flight' in step else step.delete_flight.flight_name
+            ))
+
+        if 'inject_flight' in step:
+            print("[SCD]     Step: Inject flight {} to {}".format(step.inject_flight.name, target.name))
+            resp = target.inject_flight(step.inject_flight, dry=dry)
+            # TODO: Move evaluation at the end of the execution
+            TestRunner.evaluate_inject_flight_response(step.inject_flight, resp, dry=dry)
+        elif 'delete_flight' in step:
+            print("[SCD]     Step: Delete flight {} to {}".format(step.delete_flight.flight_name, target.name))
+            target.delete_flight(step.delete_flight.flight_name, dry=dry)
+        else:
+            print("[SCD] Warning: no action defined for step {}".format(step.name))
+
+    def get_managing_target(self, flight_name: str) -> typing.Optional[TestTarget]:
+        """Returns the managing target which created a flight"""
+        for role, target in self.targets.items():
+            if target.is_managing_flight(flight_name):
+                return target
+        return None
+
+    def get_target(self, step: TestStep) -> typing.Optional[TestTarget]:
+        """Returns the target which should be called in the TestStep"""
+        if 'inject_flight' in step:
+            return self.targets[step.inject_flight.injection_target.uss_role]
+        elif 'delete_flight' in step:
+            return self.get_managing_target(step.delete_flight.flight_name)
+        else:
+            raise NotImplementedError("Unsupported step. A Test Step shall contain either a inject_flight or a delete_flight object.")
+
+    # TODO: Use this method as a canvas to create findings and move the evaluation at the end of or outside the execution.
+    @staticmethod
+    def evaluate_inject_flight_response(attempt: FlightInjectionAttempt, resp: InjectFlightResponse, dry=False) -> bool:
```

Comment:
I'd recommend not conditioning logic here on whether it's a "dry run"

Reply:
No more dry run.